etcd/src/main/scala/io/buoyant/etcd/Key.scala

```diff
@@ -0,0 +1,210 @@
+package io.buoyant.etcd
+
+import com.twitter.finagle.{Path, Service}
+import com.twitter.finagle.http._
+import com.twitter.io.Buf
+import com.twitter.util._
+import scala.collection.JavaConverters._
+
+class Key(key: Path, client: Service[Request, Response]) {
+
+  import Etcd._
+
+  def path: Path = key
```

Comment:
The library deals in finagle primitives (services, futures, etc), so I'm not sure how severe of a coupling this is.  I suppose this puts some additional requirements on permitted characters.  This all came from a Namer implementation, so it was expected it would receive paths.  I think we'd want to maintain support for passing paths to it, though it may be better to support Seq[String] as the underlying primitive.